src/ocf_seq_cutoff.c

```diff
@@ -60,25 +60,38 @@ static struct ocf_rb_node *ocf_seq_cutoff_stream_list_find(
 	return &max_stream->node;
 }
 
-void ocf_core_seq_cutoff_init(ocf_core_t core)
+int ocf_core_seq_cutoff_init(ocf_core_t core)
 {
 	struct ocf_seq_cutoff_stream *stream;
 	int i;
 
 	ocf_core_log(core, log_info, "Seqential cutoff init\n");
-	env_rwlock_init(&core->seq_cutoff.lock);
-	ocf_rb_tree_init(&core->seq_cutoff.tree, ocf_seq_cutoff_stream_cmp,
+
+	core->seq_cutoff = env_vmalloc(sizeof(*core->seq_cutoff));
+	if (!core->seq_cutoff)
+		return -OCF_ERR_NO_MEM;
+
+	env_rwlock_init(&core->seq_cutoff->lock);
+	ocf_rb_tree_init(&core->seq_cutoff->tree, ocf_seq_cutoff_stream_cmp,
 			ocf_seq_cutoff_stream_list_find);
-	INIT_LIST_HEAD(&core->seq_cutoff.lru);
+	INIT_LIST_HEAD(&core->seq_cutoff->lru);
 
 	for (i = 0; i < OCF_SEQ_CUTOFF_MAX_STREAMS; i++) {
-		stream = &core->seq_cutoff.streams[i];
+		stream = &core->seq_cutoff->streams[i];
 		stream->last = 4096 * i;
 		stream->bytes = 0;
 		stream->rw = 0;
-		ocf_rb_tree_insert(&core->seq_cutoff.tree, &stream->node);
-		list_add_tail(&stream->list, &core->seq_cutoff.lru);
+		ocf_rb_tree_insert(&core->seq_cutoff->tree, &stream->node);
+		list_add_tail(&stream->list, &core->seq_cutoff->lru);
 	}
+
+	return 0;
+}
+
+void ocf_core_seq_cutoff_deinit(ocf_core_t core)
+{
+	env_rwlock_destroy(&core->seq_cutoff->lock);
```

Comment:
I think checking whether seq_cutoff is NULL is required here (ocf_core_seq_cutoff_deinit might be called from _ocf_mngt_close_all_uninitialized_cores due to allocation failure)

Reply:
Fixed.